analytics/analytics.go

```
@@ -70,80 +89,72 @@ func InitializeAnalytics(analyticsToken, memphisV, customDeploymentId string) er
+func SendEvent(tenantName, username string, params []EventParam, eventName string) {
 	distinctId := deploymentId
 	if configuration.DEV_ENV != "" {
 		distinctId = "dev"
 	}
 
-	tenantName = strings.ReplaceAll(tenantName, "-", "_") // for parsing purposes
-	if tenantName != "" && username != "" {
-		distinctId = distinctId + "-" + tenantName + "-" + username
+	if eventName != "error" {
+		tenantName = strings.ReplaceAll(tenantName, "-", "_") // for parsing purposes
+		if tenantName != "" && username != "" {
+			distinctId = distinctId + "-" + tenantName + "-" + username
+		}
 	}
 
-	p := posthog.NewProperties()
-	p.Set("memphis-version", memphisVersion)
-
-	go AnalyticsClient.Enqueue(posthog.Capture{
-		DistinctId: distinctId,
-		Event:      eventName,
-		Properties: p,
-	})
-}
-
-func SendEventWithParams(tenantName, username string, params []EventParam, eventName string) {
-	distinctId := deploymentId
-	if configuration.DEV_ENV != "" {
-		distinctId = "dev"
+	var eventMsg []byte
+	var event *EventBody
+	var err error
+
+	if eventName == "error" {
+		event = &EventBody{
+			DistinctId:     distinctId,
+			Event:          "error",
+			Properties:     params,
+			TimeStamp:      time.Now(),
+			MemphisVersion: memphisVersion,
+		}
+	} else {
+		event = &EventBody{
+			DistinctId:     distinctId,
+			Event:          eventName,
+			Properties:     params,
+			TimeStamp:      time.Now(),
+			MemphisVersion: memphisVersion,
+		}
 	}
 
-	tenantName = strings.ReplaceAll(tenantName, "-", "_") // for parsing purposes
-	if tenantName != "" && username != "" {
-		distinctId = distinctId + "-" + tenantName + "-" + username
+	eventMsg, err = json.Marshal(event)
+	if err != nil {
+		return
 	}
-
-	p := posthog.NewProperties()
-	for _, param := range params {
-		p.Set(param.Name, param.Value)
+	err = sdkConnection.Produce("users-traces", "producer_users_traces", eventMsg, []memphis.ProducerOpt{memphis.ProducerGenUniqueSuffix()}, []memphis.ProduceOpt{})
+	if err != nil {
```

you can simply ignore errors you have 
if => return else => return